src/txn/txn_timestamp.c

```diff
@@ -280,11 +287,11 @@ __txn_global_query_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *tsp, cons
          * Compare with the least recently durable transaction.
          */
         __wt_readlock(session, &txn_global->durable_timestamp_rwlock);
-        TAILQ_FOREACH (txn, &txn_global->durable_timestamph, durable_timestampq) {
-            if (txn->clear_durable_q)
+        TAILQ_FOREACH (txn_state, &txn_global->durable_timestamph, durable_timestampq) {
```

Comment:
It's actually possible to iterate over shared transactions by traversing `txn_global->states`. However, I think in some cases the order of these queues is significant so this is a non-trivial change. For now, I've just made the queues contain `WT_TXN_SHARE` instead of `WT_TXN`.